src/app/common/theme-provider.tsx

```diff
@@ -0,0 +1,74 @@
+import {
+  createContext,
+  Dispatch,
+  FC,
+  SetStateAction,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+
+type ComputedTheme = 'dark' | 'light';
+export type UserSelectedTheme = ComputedTheme | 'system';
+
+export const labelMap = new Map<UserSelectedTheme, string>([
```

Comment:
`themeLabelMap` would help readability when used outside this file (were it not exported, labelMap would probs be fine)